### PR TITLE
docs(README): Add complex usage case related to react-i18next

### LINKED ISSUE #6: Discussion in locale change detection
## Overview

About detecting of language change in react component, storybook-addon-i18n depends on `providerLocaleKey` in a storybook parameter. when a developer use [react-intl](https://github.com/formatjs/react-intl), i18n addon can easily work. but about react-i18next, [it doesn't support props like `locale` except `i18n`](https://react.i18next.com/latest/i18nextprovider#i-18-nextprovider-props). So, additional boilerplate codes are needed.

I think it's better to choice adding new API (`onChanged` detection handler?) or update the documentation in README's complex usage.

## Possible Directions

You can see [the codes about each direction](https://github.com/taehwanno/storybook-addon-i18n-feature-request/blob/master/.storybook/config.js#L12-L51)

### 1. Add new API (`onChanged`, the name can be changed)

You can see [implementation in taehwanno/storybook-addin-i18n](https://github.com/taehwanno/storybook-addon-i18n/commit/6fd2b69ca2a4d30413e348e3f29bdd17f4a1892b)

```js
// .storybook/config.js

import { I18nextProvider } from 'react-i18next' ;
import { addParameters } from '@storybook/react';
import i18n from '../src/i18n';

addParameters({
  i18n: {
    provider: I18nextProvider,
    providerProps: {
      i18n,
    },
    providerLocaleKey: 'locale',
    supportedLocales: ['ko', 'en'],
    onChanged: (locale) => i18n.changeLanguage(locale), // <-- new API
  },
});
```

### 2. Update the documentation in README's complex usage

- Provider wrapper workaround

```js
// I18nProviderWrapper.jsx

import React, { useEffect } from 'react';
import { I18nextProvider } from 'react-i18next';

function I18nProviderWrapper({ children, i18n, locale }) {
  useEffect(() => {
    i18n.changeLanguage(locale); // <-- Change language with React effect hook
  }, [i18n, locale]);
  return <I18nextProvider i18n={i18n}>{children}</I18nextProvider>;
}

export default I18nProviderWrapper;
```

```js
// .storybook/config.js

import { addParameters } from '@storybook/react';

import i18n from '../src/i18n';
import I18nProviderWrapper from './I18nProviderWrapper';

addParameters({
  i18n: {
    provider: I18nProviderWrapper, // <-- wrapped i18n provider
    providerProps: {
      i18n,
    },
    providerLocaleKey: 'locale',
    supportedLocales: ['ko', 'en'],
  },
});
```

- `getDirection` workaround

[`getDirection` is called every re-render](https://github.com/goooseman/storybook-addon-i18n/blob/master/src/Provider.tsx#L43-L45), we can use this as a hook for language detection. but I think this isn't a proper direction given the original purpose of `getDirection` API.

```js
// .storybook/config.js

import { I18nextProvider } from 'react-i18next' ;
import { addParameters } from '@storybook/react';
import i18n from '../src/i18n';

addParameters({
  i18n: {
    provider: I18nextProvider,
    providerProps: {
      i18n,
    },
    providerLocaleKey: 'locale',
    supportedLocales: ['ko', 'en'],
    getDirection: locale => {
      i18n.changeLanguage(locale); // <-- Change language based on React render phase
      return 'ltr';
    }
  },
});
```

## Preferred Direction

I'm fine either way, whether in the direction of adding a new API or updating README documentation for a provider wrapper. Please give me an opinion :)

### PR DESCRIPTION
**Pull request by taehwanno**
Fix #6

I add some usage in README. please take a look :)

[[Markdown rendered view]](https://github.com/taehwanno/storybook-addon-i18n/blob/taehwanno%2Fdocs-add-complex-usage/README.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -121,3 +123,33 @@ addParameters({
   },
 });
 ```
+
+### react-i18next
+
+Currently, [react-i18next doesn't support other props like `locale` except `i18n`](https://react.i18next.com/latest/i18nextprovider#i-18-nextprovider-props). If you want to use i18n storybook addon, you need to wrap `I18nProvider` using with effect hook.
```
Comment (alexeychikk) [inline README.md]: > you need to wrap `I18nProvider` using with effect hook

I'm not a native speaker but I think you should change this to 

> you need to wrap `I18nProvider` with useEffect hook

Comment (taehwanno) [reply README.md]: I'm not a native speaker, too. thanks for the feedback (b643515)

File: README.md [hunk 2]
```diff
@@ -121,3 +123,33 @@ addParameters({
...
+```jsx
+export function I18nProviderWrapper({ children, i18n, locale }) {
+  React.useEffect(() => {
+    i18n.changeLanguage(locale);
+  }, [i18n, locale]);
+  return <I18nProvider i18n={i18n}>{children}</I18nProvider>;
+}
+```
+
+This Provider wrapper should accept `providerLocaleKey` as props in storybook decorator parameter. If `providerLocaleKey` prop's value is changed, [we can programmatically change the language.](https://www.i18next.com/overview/api#changelanguage)
```
Comment (alexeychikk) [inline README.md]: I think passing `providerLocaleKey: 'locale'` is redundant as it has `'locale'` value by default which is already described in the README.

And I don't understand this part:
> If `providerLocaleKey` prop's value is changed, [we can programmatically change the language.](https://www.i18next.com/overview/api#changelanguage)

`providerLocaleKey` is something static that you pass once.

Comment (taehwanno) [reply README.md]: > I think passing providerLocaleKey: 'locale' is redundant as it has 'locale' value by default which is already described in the README.

I will remove this.

> And I don't understand this part:
> 
> > If `providerLocaleKey` prop's value is changed, [we can programmatically change the language.](https://www.i18next.com/overview/api#changelanguage)
> 
> `providerLocaleKey` is something static that you pass once.

Ahh, I wrote it wrong. I wanted to imply changing the props value corresponding to providerLocaleKey. I will fix this.

Comment (taehwanno) [reply README.md]: Fixed by 098a750

File: README.md [hunk 3]
```diff
@@ -121,3 +123,32 @@ addParameters({
...
+```jsx
+export function I18nProviderWrapper({ children, i18n, locale }) {
+  React.useEffect(() => {
+    i18n.changeLanguage(locale);
+  }, [i18n, locale]);
+  return <I18nProvider i18n={i18n}>{children}</I18nProvider>;
+}
+```
+
+This Provider wrapper should accept `providerLocaleKey` as props in storybook decorator parameter. If the props value corresponding to `providerLocaleKey` is changed, [we can programmatically change the language.](https://www.i18next.com/overview/api#changelanguage)
```
Comment (alexeychikk) [inline README.md:140]: Could you please also remove this line because I don't really understand what do you mean by this.

Comment (alexeychikk) [reply README.md:140]: I think there is enough said about `providerLocaleKey` in the readme so this is redundant.

**Review by goooseman — APPROVED** [review]

**Pull request merged**